src/main/java/seedu/address/model/task/TaskList.java

```diff
@@ -113,20 +115,41 @@ public void setTask(Task taskToEdit, Task editedTask) {
         return tasksUnmodifiable;
     }
 
-    public List<Task> getSortedList(Comparator<Task> comparator) {
+    public static List<Task> getSortedList(List<Task> tasks, Comparator<Task> comparator) {
         return tasks
                     .stream()
                     .sorted(comparator)
                     .collect(Collectors.toList());
     }
 
-    public List<Task> getFilteredList(Predicate<Task> predicate) {
+    public static List<Task> getFilteredList(List<Task> tasks, Predicate<Task> predicate) {
         return tasks
                     .stream()
                     .filter(predicate)
                     .collect(Collectors.toList());
     }
 
+    /**
+     * Creates a driver list out of task list.
+     * NOTE: task list must be filtered by assigned tasks so that all the tasks contains a driver.
+     *
+     * @param assignedTasks list of task that is assigned to drivers.
+     * @return driver list that contains only drivers with assigned tasks.
+     */
+    public static List<Driver> getDriversFromTasks(List<Task> assignedTasks) {
+        List<Driver> driverList = new ArrayList<>();
+        for (Task task : assignedTasks) {
+            assert task.getDriver().isPresent();
+
+            Driver driver = task.getDriver().get();
+            if (!driverList.contains(driver)) {
```

Comment:
`driverList` then can be a set and converted to list before you return

Reply:
good way!